src/Queue.php

```diff
@@ -103,7 +108,7 @@ public function ttr(int $value): self
      *
      * @return $this
      */
-    public function delay(int $value): self
+    public function withDelay(int $value): self
```

Comment:
WDYT about moving `withDelay`, `withPriority` and `ttr` to different Job interfaces? We will can make `canPush` in the `DriverInterface`, but I'm not sure about use cases for that.

Reply:
What do you mean by "different" job interfaces?

Reply:
- `DelayableJobInterface`
- `PrioritisedJobInterface`
- `ReservableJobInterface`

Reply:
That makes sense only if these will be used separately. Would they?

Reply:
Not all drivers will support at least delayable and prioritised jobs. Will check the same for job reservation tomorrow or next week.